drivers/i2s/i2s_ll_stm32.c

```diff
@@ -603,6 +659,33 @@ static void dma_tx_callback(const struct device *dma_dev, void *arg,
 		goto tx_disable;
 	}
 
+	/*
+	 * Check if the tx stop command can be handled
+	 * if True, the tx stop command can be handled
+	 */
+	if ((stream->state == I2S_STATE_STOPPING) && (status == DMA_STATUS_COMPLETE)) {
+		/*
+		 * Check if all tx samples have been completely handled
+		 * as stated in zephyr i2s specification, in case of DRAIN command
+		 * send all data in the transmit queue and stop the transmission.
+		 */
+		if (queue_is_empty(&stream->mem_block_queue) == true) {
+			LOG_INF("All the data has been transferred");
+			stream->queue_drop(stream);
+			stream->state = I2S_STATE_READY;
+			goto tx_disable;
+		} else {
```

Comment:
Merge nested `else`+`if` into `else if` for improved readability.
Also add a comment like:
```c
/* else: DRAIN trigger -> continue TX normally until queue is empty */
```